### PR TITLE
add build and push to ghcr similar to danger

### PR DESCRIPTION
**Pull request by wilkyd**
Copy of https://github.com/danger/danger/pull/1445 & 1446 fix for JS

### CODE REVIEW

**Comment by orta** [comment]
Thanks

**Comment by orta** [comment]
I'm going to revert this as it is a breaking change - we were shipping to docker before and this removes that feature for people who were using it 

I'm open to a version which ships to both, but not breaking folks' existing workflows

**Comment by wilkyd** [comment]
Not sure I understand, both the original workflow and the updated workflow seem to solely push the Docker image to the GitHub Container Registry (ghcr.io). I don't see any push to Docker Hub or other registries. Could you please specify the breaking change? Thanks

File: .github/workflows/publish_package.yml [hunk 1]
```diff
@@ -2,27 +2,46 @@ name: Release Danger-JS package
 
 on:
   push:
+    branches:
+      - 'master'
```
Comment (glensc) [inline .github/workflows/publish_package.yml:6]: the branch is wrong., this repo has "main" as the default branch

Comment (glensc) [reply .github/workflows/publish_package.yml:6]: proposed fix: https://github.com/danger/danger-js/pull/1396

**Comment by orta** [comment]
Hah, sorry,  I misread the older build logs!

**Pull request merged**